sdcm/cluster.py

```diff
@@ -3490,16 +3491,22 @@ def execute_cmd(cql_session, entity_type):
 
         return list(regular_table_names - materialized_view_table_names)
 
-    def is_table_has_data(self, session, table_name: str) -> bool:
+    def is_table_has_data(self, session, table_name: str) -> (bool, Optional[Exception]):
         """
-        Return True if `table_name` has data in it
+        Return True if `table_name` has data in it.
+        Checking if a result page with any data is received.
         """
-        current_rows = 0
+        error = None
         try:
-            current_rows = session.execute(f"SELECT * FROM {table_name} LIMIT 1").current_rows
+            query = f"SELECT * FROM {table_name}"
+            statement = SimpleStatement(query, fetch_size=10)
+            for paged_result in session.execute(statement):
+                return bool(len(paged_result.current_rows)), None
```

Comment:
there's no need for a loop, we care only for the first result.
```suggestion
            result = session.execute(SimpleStatement(f"SELECT * FROM {table_name}", fetch_size=10))
            return bool(len(result.one()), None
```

Reply:
fixed and tested in:
https://jenkins.scylladb.com/job/scylla-staging/job/yarongilor/job/twcs-paged-query/3/

Reply:
Test ran ok and nemesis passed ( 70 times)
```
ubuntu@ip-10-12-1-17:~/sct-results/latest$ grep sdcm.nemesis.RebuildStreamingErrMonkey: sct.log | grep duration | grep -i -v "'error':" -c
73
```